server/src/main/java/org/elasticsearch/action/admin/indices/flush/TransportShardFlushAction.java

```diff
@@ -68,6 +69,11 @@ public TransportShardFlushAction(
         );
     }
 
+    @Override
+    protected ClusterBlockLevel globalBlockLevel() {
+        return ClusterBlockLevel.METADATA_READ;
+    }
```

Comment:
Surprised this wasn't checked already, today we just return a trivial success before state recovery:

```
{
  "_shards" : {
    "total" : 0,
    "successful" : 0,
    "failed" : 0
  }
}
```

We could leave this as-is ofc.